### ISSUE TITLE
Dispatch problem

### ISSUE THREAD

**Issue by jwnys** [netket/netket_fidelity#9]
I can't run netket fidelity with the latest version of netket. The problem seems to be related to dispatching:

`
    fid_opt.run(n_iter=n_iter, show_progress=show_inner_progress, out=inner_logger, callback=inner_callback())
  File "xxx/netket_fidelity/driver/infidelity_optimizer.py", line 187, in run
    super().run(n_iter, out, *args, callback=callbacks, **kwargs)
  File "xxx/netket/driver/abstract_variational_driver.py", line 255, in run
    for step in self.iter(n_iter, step_size):
  File "xxx/netket/driver/abstract_variational_driver.py", line 167, in iter
    dp = self._forward_and_backward()
  File "xxx/netket_fidelity/driver/infidelity_optimizer.py", line 145, in _forward_and_backward
    self._loss_stats, self._loss_grad = self.state.expect_and_grad(self._I_op)
  File "xxx/netket/vqs/mc/mc_state/state.py", line 604, in expect_and_grad
    return expect_and_grad(
  File "xxx/netket/utils/plum/function.py", line 366, in __call__
    return _convert(method(*args, **kw_args), return_type)
  File "xxx/netket/vqs/mc/mc_state/expect_grad.py", line 58, in expect_and_grad_default_formula
    Ō, Ō_grad = expect_and_forces(vstate, Ô, chunk_size, *args, mutable=mutable)

  File "xxx/netket/utils/plum/function.py", line 366, in __call__

    return _convert(method(*args, **kw_args), return_type)

  File "xxx/netket/vqs/mc/mc_state/expect_forces.py", line 48, in expect_and_forces
    σ, args = get_local_kernel_arguments(vstate, Ô)

  File "xxx/netket/utils/plum/function.py", line 365, in __call__
    method, return_type = self._resolve_method_with_cache(args=args)
  File "xxx/netket/utils/plum/function.py", line 396, in _resolve_method_with_cache
    method, return_type = self.resolve_method(args)
  File "xxx/netket/utils/plum/function.py", line 316, in resolve_method
    impl, return_type = self._handle_not_found_lookup_error(e)
  File "xxx/netket/utils/plum/function.py", line 325, in _handle_not_found_lookup_error
    raise ex from None
  File "xxx/netket/utils/plum/function.py", line 298, in resolve_method
    method = self._resolver.resolve(target)
  File "xxx/netket/utils/plum/resolver.py", line 293, in resolve
    raise NotFoundLookupError(self.function_name, target, self.methods)
`

it can't find the expect_and_grad of the type:
`
netket.utils.plum.resolver.NotFoundLookupError: 
get_local_kernel_arguments(MCState, op) could not be 
resolved.
`
where op is an existing netket operator.
This does work if I go back to the netket version before:
https://github.com/netket/netket/commit/34bd4adde13df35b65f4f055a750dda242fdfa20

**Comment by PhilipVinc** [comment]
Yes. NetKet will change the dispatch interface in the **next** release. 

Contributions to adapt to the changes would be welcome.
